test/extended/etcd/helpers/helpers.go

```diff
@@ -180,6 +183,91 @@ func recoverClusterToInitialStateIfNeeded(ctx context.Context, t TestingT, machi
 	})
 }
 
+func DeleteSingleMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
+
+	machineToDelete := ""
+	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+		if err != nil {
+			return isTransientAPIError(t, err)
+		}
+
+		// Machine names are suffixed with an index number (e.g "ci-op-xlbdrkvl-6a467-qcbkh-master-0")
+		// so we sort to pick the lowest index, e.g master-0 in this example
+		machineNames := []string{}
+		for _, m := range machineList.Items {
+			machineNames = append(machineNames, m.Name)
+		}
+		sort.Strings(machineNames)
+		machineToDelete = machineNames[0]
+		t.Logf("attempting to delete machine %q", machineToDelete)
+
+		if err := machineClient.Delete(ctx, machineToDelete, metav1.DeleteOptions{}); err != nil {
+			// The machine we just listed should be present but if not, error out
+			if apierrors.IsNotFound(err) {
+				t.Logf("machine %q was listed but not found or already deleted", machineToDelete)
+				return false, fmt.Errorf("machine %q was listed but not found or already deleted", machineToDelete)
```

Comment:
shouldn't this be:

> return true, null

after all, the expectation of the whole function is that you want to delete the machine? :dagger:

Reply:
Yes we want to delete the machine but having just listed the machine in L93 and then subsequently finding out the machine is no longer present means something has gone wrong.

Deleting the machine doesn't mean it should just go away immediately. You need the replacement machine to be created, added, and promoted as a member, and then have member removal and deletion hook removal etc before the machine should go away.

If at this point deleting the the machine results in an `IsNotFound ` error then the test really can't verify anything  anymore in above sequence for the vertical scaling workflow since the machine is already gone.